pkg/operator/cassandra/controller/util/resource.go

```diff
@@ -68,6 +68,10 @@ func StatefulSetForRack(r cassandrav1alpha1.RackSpec, c *cassandrav1alpha1.Clust
 			Namespace:       c.Namespace,
 			Labels:          rackLabels,
 			OwnerReferences: []metav1.OwnerReference{NewControllerRef(c)},
+			Annotations: map[string]string{
+				"prometheus.io/scrape": "true",
```

Comment:
The Prometheus annotations should be set at the PodSpec level and not Deployment level AFAIK. So in this case moved to line 88.

Would be good if you could verify that the Prometheus annotations are at the right place. You could verify that with, e.g., the prometheus-operator to see if the Prometheus picks the Pods up as wanted.

Reply:
Ok, I will change it.

Reply:
Prometheus operator won't work with scrape annotations. You'd have to create a service monitor for it to be picked up.
I think that's out of scope (but an example could be included in the readme)